metasynth/provider.py

```diff
@@ -0,0 +1,456 @@
+
+        Returns
+        -------
+        BaseDistribution:
+            Distribution representing the dictionary.
+        """
+        for dist_class in self.get_dist_list(var_dict["type"]):
+            if dist_class.implements == var_dict["distribution"]["implements"]:
+                return dist_class.from_dict(var_dict["distribution"])
+        raise ValueError(f"Cannot find distribution with name "
+                         f"'{var_dict['distribution']['implements']}'"
+                         f"and type '{var_dict['type']}'.")
+
+
+class BuiltinDistributionProvider(BaseDistributionProvider):
+    """Distribution tree that includes the builtin distributions."""
+
+    @property
+    def name(self):
+        return "builtin"
+
+    @property
+    def version(self):
+        return "1.0"
+
+    @property
+    def distributions(self) -> list[type[BaseDistribution]]:
+        return [
+            DiscreteUniformDistribution, PoissonDistribution, UniqueKeyDistribution,
+            UniformDistribution, NormalDistribution, LogNormalDistribution,
+            TruncatedNormalDistribution, ExponentialDistribution,
+            MultinoulliDistribution,
+            RegexDistribution, UniqueRegexDistribution, FakerDistribution,
+            UniformDateDistribution,
+            UniformTimeDistribution,
+            UniformDateTimeDistribution,
+        ]
+
+
+class DistributionProviderList():  # pylint: disable=too-few-public-methods
+    """List of DistributionProvider's with functionality to fit distributions.
+
+    Arguments
+    ---------
+    dist_providers:
+        One or more distribution providers, that are denoted either with a string ("builtin")
+        , DistributionProvider (BuiltinDistributionProvider()) or DistributionProvider type
+        (BuiltinDistributionProvider).
+        The order in which distribution providers are included matters. If a provider implements
+        the same distribution at the same privacy level, then only the first will be taken into
+        account.
+    """
+
+    def __init__(
+            self,
+            dist_providers: Union[
+                str, type[BaseDistributionProvider], BaseDistributionProvider,
+                list[Union[str, type[BaseDistributionProvider], BaseDistributionProvider]]]):
+        if isinstance(dist_providers, (str, type, BaseDistributionProvider)):
+            dist_packages = [dist_providers]
+        self.dist_packages = []
+        for pkg in dist_packages:
+            if isinstance(pkg, str):
+                self.dist_packages.append(get_distribution_provider(pkg))
+            elif isinstance(pkg, type):
+                self.dist_packages.append(pkg())
+            elif isinstance(pkg, BaseDistributionProvider):
+                self.dist_packages.append(pkg)
+            else:
+                raise ValueError(f"Unknown distribution package type '{type(pkg)}'")
+
+    def fit(self, series: pl.Series,  # pylint: disable=too-many-arguments
```

Comment:
I don't like these comments. Let's configure pylint globally in `pyproject.toml`